client/wfnews-war/src/main/angular/src/background.js

```diff
@@ -0,0 +1,111 @@
+const Storage = require('@ionic/storage');
+
+function submitOfflineRoF ( event ) {
+  addEventListener(event, (resolve, reject) => { 
+    try {
+        scheduleDataSync();
+        resolve();
+    } catch (err) {
+        reject(err);
+    }
+  });
+}
+
+ async function scheduleDataSync() {
+    const storage = new Storage();
+    await storage.create();
+    const isConnected = await checkOnlineStatus();
+      if (isConnected) {
+        await syncDataWithServer();
+      }
+    }
+
+ async function syncDataWithServer() {
+  const storage = new Storage();
+    try {
+      // Fetch and submit locally stored data
+      const offlineReport = await storage.get('offlineReportData');
+
+      if (offlineReport) {
+        // Send the report to the server
+        const response = await submitReportToServer(offlineReport);
+
+        if (response.success) {
+          // Remove the locally stored data if sync is successful
+          await storage.remove('offlineReportData');
+        }
+      }
+    } catch (error) {
+      console.error('Sync failed:', error);
+    }
+  }
+
+async function submitReportToServer(offlineReport){
```

Comment:
These functions exist elsewhere, might be good to keep this in a service class so we dont need to make identical changes in multiple places

Reply:
Note: I understand they're in the other PR, so we can likely make that change there, merge it in, then update this one to use the shared code